### PR TITLE
App updates for Formplayer

### PR DESCRIPTION
**Pull request by wpride**
Allow Formplayer to update an application instead of always deleting DBs and re-installing. PR appears strange because a lot of this code was merge but unused already; majority of code changes are error handling. 

cross-request: https://github.com/dimagi/commcare-core/pull/785

### CODE REVIEW

File: src/main/java/session/MenuSession.java [hunk 1]
```diff
@@ -127,8 +130,26 @@ private void initializeBreadcrumbs() {
...
+                return new NotificationMessage("Application updated successfully.", false);
+            } else {
+                return new NotificationMessage("Application up to date.", false);
+            }
+        } catch (UnresolvedResourceException e) {
+            String message = "Update Failed! Couldn't find or install one of the remote resources";
+            log.error(message, e);
+            return new NotificationMessage(message, true);
+        } catch (UnfullfilledRequirementsException e) {
+            String message = "Update Failed! This CLI host is incompatible with the app";
```
Comment (amstone326) [inline src/main/java/session/MenuSession.java]: This seems like a strange message to show in formplayer; I don't think we want to mention the CLI here

File: src/main/java/session/MenuSession.java [hunk 2]
```diff
@@ -127,8 +130,26 @@ private void initializeBreadcrumbs() {
...
+            log.error(message, e);
+            return new NotificationMessage(message, true);
+        } catch (Exception e) {
+            String message = "Update Failed! There is a problem with one of the resources";
```
Comment (amstone326) [inline src/main/java/session/MenuSession.java]: Should this message be more generic, given that we really have no idea where this Exception came from if we get to this point?

File: src/main/java/session/MenuSession.java (hunk 1 above)
Comment (wpride) [reply src/main/java/session/MenuSession.java]: Good call

File: src/main/java/session/MenuSession.java (hunk 2 above)
Comment (wpride) [reply src/main/java/session/MenuSession.java]: Good point; I copied this code and it looks like this catch was needlessly generic, the only typed exception remaining was `InstallCancelledException` which I'm now catching explicitly

**Comment by amstone326** [comment]
@wpride Approved but the build seems to be failing on both this and the `core` PR

**Pull request merged**